pkg/deploy/template_chart_manager.go

```diff
@@ -0,0 +1,127 @@
+// the watcher's helm chart and installs it using a native kube-client.
+type SKRWebhookTemplateChartManager struct {
+	config  *SkrChartManagerConfig
+	kcpAddr string
+}
+
+type SkrChartManagerConfig struct {
+	// WebhookChartPath represents the path of the webhook chart
+	// to be installed on SKR clusters upon reconciling kyma CRs.
+	WebhookChartPath       string
+	SkrWebhookMemoryLimits string
+	SkrWebhookCPULimits    string
+	// WatcherLocalTestingEnabled indicates if the chart manager is running in local testing mode
+	WatcherLocalTestingEnabled bool
+	// GatewayHTTPPortMapping indicates the port used to expose the KCP cluster locally for the watcher callbacks
+	GatewayHTTPPortMapping int
+}
+
+func NewSKRWebhookTemplateChartManager(kcpRestConfig *rest.Config, config *SkrChartManagerConfig,
+) (*SKRWebhookTemplateChartManager, error) {
+	resolvedKcpAddr, err := resolveKcpAddr(kcpRestConfig, config)
+	if err != nil {
+		return nil, err
+	}
+	return &SKRWebhookTemplateChartManager{
+		config:  config,
+		kcpAddr: resolvedKcpAddr,
+	}, nil
+}
+
+func (m *SKRWebhookTemplateChartManager) Install(ctx context.Context, kyma *v1alpha1.Kyma) (bool, error) {
+	logger := logf.FromContext(ctx)
+	kymaObjKey := client.ObjectKeyFromObject(kyma)
+	syncContext := remote.SyncContextFromContext(ctx)
+	chartArgValues, err := generateHelmChartArgs(ctx, syncContext.ControlPlaneClient, kymaObjKey, m.config, m.kcpAddr)
+	if err != nil {
+		return true, err
+	}
+	manifest, err := renderChartToRawManifest(ctx, kymaObjKey, m.config.WebhookChartPath, chartArgValues)
+	if err != nil {
+		return true, err
+	}
+	resources, err := getRawManifestUnstructuredResources(manifest)
+	if err != nil {
+		return true, err
+	}
+	errGrp, grpCtx := errgroup.WithContext(ctx)
+	for idx := range resources {
+		resIdx := idx
+		errGrp.Go(func() error {
+			return syncContext.RuntimeClient.Patch(grpCtx, resources[resIdx], client.Apply,
+				client.ForceOwnership, skrChartFieldOwner)
+		})
+	}
+	if err := errGrp.Wait(); err != nil {
+		return true, err
+	}
+	kyma.UpdateCondition(v1alpha1.ConditionReasonSKRWebhookIsReady, metav1.ConditionTrue)
+	logger.Info("successfully installed webhook chart",
+		"release-name", skrChartReleaseName(kymaObjKey))
+	return false, nil
+}
+
+func (m *SKRWebhookTemplateChartManager) Remove(ctx context.Context, kyma *v1alpha1.Kyma) error {
+	logger := logf.FromContext(ctx)
+	kymaObjKey := client.ObjectKeyFromObject(kyma)
+	syncContext := remote.SyncContextFromContext(ctx)
+	manifest, err := renderChartToRawManifest(ctx, kymaObjKey, m.config.WebhookChartPath, map[string]interface{}{})
+	if err != nil {
+		return err
+	}
+	resources, err := getRawManifestUnstructuredResources(manifest)
+	if err != nil {
+		return err
+	}
+	for _, resource := range resources {
+		if err := syncContext.RuntimeClient.Delete(ctx, resource); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete webhook %s: %w", resource.GetKind(), err)
+		}
+	}
```

Comment:
does it make sense to implement similar errGrp here, so that deletion of following resources are not blocked by one failing?